docs/Java-SpringBoot-Developer.md

```diff
@@ -0,0 +1,355 @@
+
+- Open the application.properties file and add below properties to specify your Azure Key Vault url
+- azure.keyvault.enabled is used to turn on/off Azure Key Vault Secret property source, default is true.
+
+[application.properties](https://github.com/microsoft/helium-java/blob/master/src/main/resources/application.properties)
+
+```properties
+azure.keyvault.enabled=true
+azure.keyvault.uri=https://${KeyVaultName}.vault.azure.net/
+```
+
+### JAVA-SPRINGBOOT-KEYVAULT-SDK-GAP : Local Development issue with spring-boot starter for key-vault with MSI
+
+- This does not works in the local development scenario as the spring-boot keyvault java sdk fails to get Key Vault access through MSI on local development environment​
+- This is a security issue that affects the development environment
+
+Local development environment can only access keyvault with clear-text credentials as below
+
+```properties
+
+azure.keyvault.uri=https://${KeyVaultName}.vault.azure.net/
+azure.keyvault.client-id=${client_id}
+azure.keyvault.client-key=${client_key}
+
+```
+
+#### Solution
+
+1. Use clear-text for testing locally
+2. Use service principal with client-key stored in the Azure DevOps as secret value
+
+#### Now, you can get Azure Key Vault secret value as a configuration property in spring framework
+
+[Application.java](https://github.com/microsoft/helium-java/blob/master/src/main/java/com/microsoft/azure/helium/Application.java)
+
+```java
+
+@EnableSwagger2
+@SpringBootApplication
+public class Application implements CommandLineRunner {
+    private static final Logger logger = LoggerFactory.getLogger(Application.class);
+    @Autowired
+    Environment environment;
+
+    @Value("${azure.keyvault.uri}")
+    private String keyUri;
+
+    @Value("${azure.keyvault.client-key}")
+    private String key;
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+    public void run(String... varl) throws Exception {
+        logger.info("keyUri: " + keyUri);
+        logger.info("key: " + key);
+    }
+}
+
+```
+
+## Key Rotation
+
+- Work in progress
+
+## Cosmos DB
+
+### Spring Repository Pattern
+
+### JAVA-SPRINGBOOT-COSMOSDB-SDK-GAP: Spring framework does not support different document types in the same collection
+
+This is limitation and needs a spring-boot patch. Spring-Boot entity framework works on entity<-> repo mapping.
+Hence we cannot model movies, genre,actors in the same "movies" collection as recommended in this best practices [document](https://github.com/4-co/imdb)
+spring-data-cosmosdb expects to have 3 separate collections as "movies", "genre", "actor"
+However this is cost implication as "genre" collection having just 30 elements needs a separate collection and is 3 times the RU querying for 3 different collection.
+
+#### Solution
```

Comment:
Solution appears as a header multiple times which causes 3 linter errors. Should we ignore the linter error, not use a heading or distinguish the headings?

Reply:
How about remove the "####" heading and make "Solution" **bold** ?